### PR TITLE
Rewrite to be compat of base's Data.Type.Equality

### PR DESCRIPTION
**Pull request by phadej**
EDIT: the travis build from my fork is https://travis-ci.org/phadej/type-equality/builds/578721603 = all green

### CODE REVIEW

File: type-equality.cabal [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
-  Exposed-Modules:     Data.Type.Equality
+tested-with:
+  GHC ==8.8.1 || ==8.6.5 || ==8.4.4 || ==8.2.2 || ==8.0.2 || ==7.10.3 || ==7.8.4 || ==7.6.3 || ==7.4.2 || ==7.2.2 || ==7.0.4
 
-Source-Repository head
-  Type:                git
-  Location:            git://github.com/hesselink/type-equality.git
+source-repository head
+  type:     git
+  location: git://github.com/phadej/some.git
```
Comment (hesselink) [inline type-equality.cabal]: Can you change this to my repo?

**Comment by phadej** [comment]
@hesselink thanks for your comments, hopefully I noticed everything

File: LICENSE [hunk 2]
```diff
@@ -1,28 +1,30 @@
-Copyright (c) Erik Hesselink 2009
+Copyright (c) 2009 Erik Hesselik, 2019 Oleg Grenrus, Ryan Scott
```
Comment (hesselink) [inline LICENSE]: Still a typo in my last name here, should be 'Hesselink' with an 'n'.

Comment (RyanGlScott) [reply LICENSE]: Fixed.

**Comment by phadej** [comment]
Thanks @RyanGlScott ,

@hesselink if everything looks great now, I can handle the release practicalities

**Comment by hesselink** [comment]
Thanks! I've added you both as contributors to this repo, and as maintainers on hackage, so you should be good to go. LMK if something is still missing.

**Comment by phadej** [comment]
Thanks Erik, the https://hackage.haskell.org/package/type-equality-1 is on Hackage now.

**Pull request merged**